src/main/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/client/RetryNetworkExceptions.kt

```diff
@@ -0,0 +1,17 @@
+package uk.gov.justice.digital.hmpps.hmppschallengesupportinterventionplanapi.client
+
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import reactor.core.publisher.Mono
+import reactor.util.retry.Retry
+import java.io.IOException
+import java.time.Duration
+
+fun <T> Mono<T>.retryNetworkExceptions(): Mono<T> =
```

Comment:
Are we confident that every endpoint we have and are going to have is idempotent, if we're retrying on things like 5xx errors? I'm unsure as to the APi architecture generally and what the plans are for it as we develop CSIP.

Reply:
This is a fair concern. The retry mechanism is only used and indeed should only be used for idempotent, read only downstream endpoints. Getting user or prisoner information in this case. It's probably worth renaming this to make it's purpose clear. Something like `retryReadOnlyRequest` or even `retryIdempotentEvent` with some function documentation